stdlib/public/core/KeyPath.swift

```diff
@@ -1764,7 +1764,12 @@ internal struct KeyPathBuffer {
     internal mutating func pushRaw(size: Int, alignment: Int)
         -> UnsafeMutableRawBufferPointer {
       var baseAddress = buffer.baseAddress.unsafelyUnwrapped
+#if os(Android) && arch(arm64)
+      // Android AArch64 may tag the first byte so remove it before checking alignment.
+      var misalign = (0x00FF_FFFF_FFFF_FFFF & Int(bitPattern: baseAddress)) % alignment
```

Comment:
Someone else should review this impact. We may want/need to tidy up our macros and builtins some to avoid a lot of error-prone `#if` logic.

Reply:
@jckarter, you added these alignment checks to this file in #10419, please let me know what you think of these two changes to avoid pointer tags.

Reply:
Masking off the top bytes should not affect the alignment computation. Is the problem here the signed-ness of `%`? How about one of these instead, which are also probably more accurate on other platforms:

```
var misalign = Int(UInt(bitPattern: baseAddress)) % alignment
```

or

```
var misalign = Int(bitPattern: baseAddress) & (alignment - 1)
```

Reply:
Or if the standard library has since grown standard APIs for pointer alignment, maybe we can adopt those here.

Reply:
@buttaface ?

Reply:
> Is the problem here the signed-ness of %?

Probably so, the memory tag on Android could be anything and right now it is always statically set to `B4` in the top byte, which was changing the result of this modulo calculation.

I'll try your alternatives and report back, don't know what methods the stdlib provides.

Reply:
> `var misalign = Int(UInt(bitPattern: baseAddress)) % alignment`

I just tried it and six stdlib test modules fail, presumably because this doesn't avoid the problem of the top sign bit being set.

> `var misalign = Int(bitPattern: baseAddress) & (alignment - 1)`

All the same stdlib tests pass when built in release mode, but this assumes the alignment is a power of two. I figured you were using the modulo before because you couldn't assume that, can you?

I ran some searches in the stdlib and while there is a `Builtin._roundUp()` function, I don't see anything that returns the misalignment, as we're calculating here.

Reply:
@jckarter I couldn't tell from the conversation, but has your concern been addressed?

Reply:
We can assume the alignment is a power of two. We should use `var misalign = Int(bitPattern: baseAddress) & (alignment - 1)` everywhere. The existing code would also be a bug on other 64-bit platforms that might have the high bit set for a valid pointer.

Reply:
Will do.

Reply:
Done.